### PR TITLE
Add macos specific options to allow different builds

### PR DESCRIPTION
**Pull request by mirageglobe**
- fix for https://github.com/Blightmud/Blightmud/issues/768

### CODE REVIEW

**Comment by mirageglobe** [comment]
need to update the SHA.
caveat. the if macos condition should work. I have no way to test it unfortunately. if not, it can be removed and just use the conditions for `on_big_sur` etc

ref : Formula/lastpass-cli.rb

**Comment by LiquidityC** [comment]
Nice. I'll ping here once there are actual artifacts to get SHA-1 sums from.

**Comment by mirageglobe** [comment]
perfect. just added info. I found its also possible to have `if OS.linux?` .. so something to note in the future. also there is a great way i found in homebrew docs. just grep the local homebrew directory > Formula for examples. 

theres a few such as delta <https://github.com/Homebrew/homebrew-core/blob/b09ed658fd1b59285e80442746bca36bb2f7d078/Formula/git-delta.rb> that is rust built locally. it seems to be much simpler than i had anticipated. something if you'd like to go the route of building locally rather than pipeline.

**Comment by LiquidityC** [comment]
If you are able to reach this link it should be a working MacOs 11 build: https://github.com/Blightmud/Blightmud/releases/download/untagged-a5c172d7755a75a5e4c6/blightmud-v5.2.0-beta.1-macos-11.zip

You can test it out with checksums and all.

I'll be making an official release this week or next. Once that is done you'll have access to both builds and you can adapt this formula accordingly and we can merge it then. Sorry about the wait. There's quick way to fiddle with releases.

File: Formula/blightmud.rb [hunk 1]
```diff
@@ -5,8 +5,18 @@ class Blightmud < Formula
   license "GPL-3.0-only"
   depends_on "openssl"
 
-  url "https://github.com/Blightmud/Blightmud/releases/download/v#{version}/blightmud-v#{version}-macos.zip"
-  sha256 "a9bb55128dd61f24bbbee4efca94f1b3dfa45d57f4abda7494a569e824332676"
+  if OS.mac?
+    on_big_sur do
+      url "https://github.com/Blightmud/Blightmud/releases/download/v#{version}/blightmud-v#{version}-macos-bigsur.zip"
```
Comment (LiquidityC) [inline Formula/blightmud.rb]: The filename will be `blightmud-v#{version}-macos-11.zip`

**Comment by LiquidityC** [comment]
Here are the correct hash sums:

    MacOS (latest) sha256: c6c325c5b6ad67fcb2b7eb7cadf63fbdda00bfae3d0ce26c50f83743ef8e5805
    MacOS (11) sha256: 8f6e089a59745afe6c07d82c9db7417e728bdab550e25db3d3c82aac8a872939

**Comment by mirageglobe** [comment]
i will take a quick look at this.

File: Formula/blightmud.rb (hunk 1 above)
Comment (mirageglobe) [reply Formula/blightmud.rb]: fixed 👍

**Comment by mirageglobe** [comment]
rebased and seems like the conflict closed this PR.. anyway the revised is here : https://github.com/Blightmud/homebrew-blightmud/pull/5 👯

**Comment by mirageglobe** [comment]
> If you are able to reach this link it should be a working MacOs 11 build: https://github.com/Blightmud/Blightmud/releases/download/untagged-a5c172d7755a75a5e4c6/blightmud-v5.2.0-beta.1-macos-11.zip
> 
> You can test it out with checksums and all.
> 
> I'll be making an official release this week or next. Once that is done you'll have access to both builds and you can adapt this formula accordingly and we can merge it then. Sorry about the wait. There's quick way to fiddle with releases.

i cant seem to see this file. link might have expired or something? but it should work.

**Pull request closed without merging**